README.md

```diff
@@ -1,25 +1,437 @@
+
+Annotations are defined inside the annotations object. This object has an array of groups, and each group element has a unique id. The groups object contains an array of items, each of which contains information on one specific annotation in the chart.
+
+Here we have created an external json file which contains the chart data and we implemented a button when clicked renders the chart with the annotation.
+
+In the Index.razor file, firstly the data is read from the json file and the chart is rendered as follows:
+```
+string jsonUrl = navigationManager.ToAbsoluteUri("demo1.json").ToString();
+dynamic myDataSource = jsonUrl;
+await xyz.renderChart(chartConfig);
+await xyz.CallFusionChartsFunction("setJSONUrl", "demoId", jsonUrl);
+```
+Now there is a user-defined method addAnnotaotion() which acts renders the chart by adding an item or a group upon button click. Here is the sample where an item is added:
+
+```
+private async Task addAnnotation(){
+        await xyz.addAnnotations("addItem", "demoId", "infobar", new {
+                id = "label1",
+                align= "RiGHT",
+                type= "text",
+                text= "Messi added{br}roughly 7 Goals from{br}his shot quality",
+                fillcolor= "#2F9AC4",
+                rotate= "90",
+                x= "$dataset.1.set.1.endx+65",
+                y= "$dataset.0.set.5.y"
+            }, true);
+    }
+```
+The flow leads to FusionChartsService.cs file and its implementation is as follows:
+```
+public async Task addAnnotations(String functionName, String id, params object[] args){
+  await _jsruntime.InvokeVoidAsync("FusionCharts.addAnnotations", functionName, id, args);
+}
+```
+Finally the user-defined addAnnotations() is invoked to add either an item or a group to the rendered chart:
+
+```
+window.FusionCharts.addAnnotations = (functionName, id, args) => {
+  let annotations = FusionCharts(id).annotations;
+  annotations[functionName].apply(annotations, args);
+};
+
+```
+
+### Cancel Events 
+
+The functionality to trigger the Cancel Events is implemented in Blazor, which disposes the event that is already triggered on the chart.
+In the Index.razor file, the functionality of the event that has to be disposed is as follows:
+
+```
+myEvent.dataPlotClick = "function() {DotNet.invokeMethodAsync('BlazorApp2TestQA2', 'ChangeData')}";
+myEvent.beforeDispose = "function(e) { console.log(e); e.preventDefault() }"; 
+myEvent.disposeCancelled = "function() { console.log('dispose cancelled') }";
+```
+
+Upon a button click, the event can be canclelled by invoking the callDispose() method whose implementation is attached below:
+```
+private async Task callDispose(){
+        await xyz.CallFusionChartsFunction("dispose", "demoId");
+}
+```
+
+In the blazor-fusioncharts.js file, there are multiple methods implemented to handle the multiple events cancellation.\
```

Comment:
this section is not needed where we are describing about extractArguments, extractBody and reviverFunction

Reply:
Done